src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Selectors/SamlSecurityTokenAuthenticator.cs

```diff
@@ -0,0 +1,243 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using CoreWCF.IdentityModel.Claims;
+using CoreWCF.IdentityModel.Policy;
+using CoreWCF.IdentityModel.Tokens;
+using CoreWCF.Security;
+using MSSaml = Microsoft.IdentityModel.Tokens.Saml;
+
+namespace CoreWCF.IdentityModel.Selectors
+{
+    public class SamlSecurityTokenAuthenticator : SecurityTokenAuthenticator
+    {
+        private List<SecurityTokenAuthenticator> _supportingAuthenticators;
+        private Collection<string> _allowedAudienceUris;
+        private AudienceUriMode _audienceUriMode;
+        private TimeSpan _maxClockSkew;
+
+        public SamlSecurityTokenAuthenticator(IList<SecurityTokenAuthenticator> supportingAuthenticators)
+            : this(supportingAuthenticators, TimeSpan.Zero)
+        { }
+
+        public SamlSecurityTokenAuthenticator(IList<SecurityTokenAuthenticator> supportingAuthenticators, TimeSpan maxClockSkew)
+        {
+
+            if (supportingAuthenticators == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(supportingAuthenticators));
+
+            this._supportingAuthenticators = new List<SecurityTokenAuthenticator>(supportingAuthenticators.Count);
+            for (int i = 0; i < supportingAuthenticators.Count; ++i)
+            {
+                this._supportingAuthenticators.Add(supportingAuthenticators[i]);
+            }
+
+            _maxClockSkew = maxClockSkew;
+            _audienceUriMode = AudienceUriMode.Always;
+            _allowedAudienceUris = new Collection<string>();
+        }
+
+        public AudienceUriMode AudienceUriMode
+        {
+            get { return _audienceUriMode; }
+            set
+            {
+                AudienceUriModeValidationHelper.Validate(_audienceUriMode);
+                _audienceUriMode = value;
+            }
+        }
+
+        public IList<string> AllowedAudienceUris
+        {
```

Comment:
Auto property with inline initialization.
```c#
public IList<string> AllowedAudienceUris { get; } = new Collection<string>();
```
Then you can remove the initialization in the constructor.